astroimtools/cutout_tools.py

```diff
@@ -16,11 +16,267 @@
+
+    Parameters
+    ----------
+    image : str or `HDUList` or `PrimaryHDU` or `ImageHDU` or `CompImageHDU`
+        Image to cut from. If string is provided, it is assumed to be a
+        fits file path.
+    catalog : str or `astropy.table.table.Table`
+        Catalog table defining the sources to cut out. Must contain
+        unit information as the cutouttool does not assume default units.
+    image_ext : int, optional
+        Image extension to extract header and data. Default is 0.
+    origin : int
+        Whether pixel coordinates are 0 or 1-basedpixel coordinates.
+    to_fits : bool
+        Save cutouts to fits files. Each cutout will be saved to a separate file.
+    output_dir : str
+        Path to directory to save the cutouts in. Set to
+        current_working_dir/cutouts by default
+    overwrite: bool, optional
+        Overwrite existing files. Default is `False`.
+    catalog_format : str, BaseReader
+        Input catalog format if reading from a file. Default is `ascii.ecsv`.
+        Note: Catalog format should be capable of staring units as metadata.
+    delimiter: str
+        Input catalog column delimiter string if reading from a file.
+    suppress_rotation : bool
+        Suppress rotation even if ``'cutout_pa'`` is provided. Default is `False`.
+    verbose : bool, optional
+        Print extra info. Default is `True`.
+
+    Returns
+    -------
+    cutouts : list
+        A list of PrimaryHDU or file names (if saving to fits).
+        If cutout failed for a target, `None` will be added as a
+        place holder.
+    """
+    # Optional dependencies...
+    from reproject.interpolation.high_level import reproject_interp
+
+    # read in the catalog file:
+    if isinstance(catalog, str):
+        if delimiter is None:
+            catalog = QTable.read(catalog, format=catalog_format)
+        else:
+            catalog = QTable.read(catalog, format=catalog_format, delimiter=delimiter)
+    elif not isinstance(catalog, Table):
+        raise TypeError("Catalog should be an astropy.table.table.Table or"
+                        " file name, got {0} instead".format(type(catalog)))
+
+    # Get data and wcs from ImageHDU:
+    if isinstance(image, str):
+        with fits.open(image) as pf:
+            image_hdu = pf[image_ext]
+            data = image_hdu.data
+            wcs = WCS(image_hdu.header)
+    else:
+        if isinstance(image, fits.hdu.hdulist.HDUList):
+            image_hdu = image[image_ext]
+        elif isinstance(image, (PrimaryHDU, ImageHDU, CompImageHDU)):
+            image_hdu = image
+        else:
+            raise TypeError("Expected ImageHDU, HDUList, or file name. Got {0} instead".format(type(image)))
+        data = image_hdu.data
+        wcs = WCS(image_hdu.header)
+
+    # Calculate the pixel scale of input image:
+    pixel_scale_width = proj_plane_pixel_scales(wcs)[0] * u.Unit(wcs.wcs.cunit[0]) / u.pix
```

Comment:
for efficiency and error checking might want to assign `proj_plane_pixel_scales(wcs)` to a variable, first. Then make sure it is not empty?  (might be more error checking than needed, but I would at least assign it to a variable.